packages/data-stores/src/site/types.ts

```diff
@@ -174,3 +174,19 @@ export interface Domain {
 	product_slug?: any;
 	owner: string;
 }
+
+export interface SiteLaunchState {
+	status: SiteLaunchStatus;
+	errorCode: SiteLaunchError | undefined;
+}
+
+export enum SiteLaunchError {
+	INTERNAL = 'internal',
+}
+
+export enum SiteLaunchStatus {
+	UNINITIALIZED = 'unintialized',
+	IDLE = 'idle',
```

Comment:
For the way we've been using `IDLE`, I believe it should be renamed to `IN_PROGRESS` or something like that.

Also because `UNINITIALIZED` already has a similar meaning.